src/sentry/api/serializers/models/environment.py

```diff
@@ -1,28 +1,52 @@
 from collections import namedtuple
 from datetime import timedelta
+from typing import List
 
 from django.utils import timezone
 
 from sentry.api.serializers import Serializer, register
 from sentry.app import tsdb
-from sentry.models import Environment, EnvironmentProject
+from sentry.models import Environment, EnvironmentBookmark, EnvironmentProject, User
 
 StatsPeriod = namedtuple("StatsPeriod", ("segments", "interval"))
 
 
+def get_bookmarks(env_list: List[Environment], user: User):
+    """
+    Gets a list of environment id's which are bookmarked by the user
+    """
+    if not user.is_authenticated or not env_list:
+        return set()
+
+    return set(
+        EnvironmentBookmark.objects.filter(
+            user=user, environment_id__in=[i.id for i in env_list]
+        ).values_list("environment_id", flat=True)
+    )
+
+
 @register(Environment)
 class EnvironmentSerializer(Serializer):
+    def get_attrs(self, item_list, user):
+        bookmarks = get_bookmarks(item_list, user)
+        return {item: {"is_bookmarked": item.id in bookmarks} for item in item_list}
+
     def serialize(self, obj, attrs, user):
-        return {"id": str(obj.id), "name": obj.name}
+        return {"id": str(obj.id), "name": obj.name, "isBookmarked": attrs["is_bookmarked"]}
 
 
 @register(EnvironmentProject)
 class EnvironmentProjectSerializer(Serializer):
+    def get_attrs(self, item_list, user):
+        bookmarks = get_bookmarks(item_list, user)
+        return {item: {"is_bookmarked": item.id in bookmarks} for item in item_list}
```

Comment:
Same re the `expand` param

Reply:
same comment as above

Reply:
Same as above - this requires an extra db call, I'd just recommend not including it by default